pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -27,59 +27,80 @@ import (
 	"k8s.io/kubernetes/pkg/kubectl"
 )
 
-// shortcutExpander is a RESTMapper that can be used for Kubernetes resources.   It expands the resource first, then invokes the wrapped
+// NoResourceWithCommendError
+type NoResourceWithCommendError struct {
+	*meta.NoResourceMatchError
+
+	Recommends []string
+}
+
+var _ error = &NoResourceWithCommendError{}
+
+// shortcutExpander is a RESTMapper that can be used for Kubernetes resources.
+// It expands the resource first, then invokes the wrapped
 type shortcutExpander struct {
 	RESTMapper meta.RESTMapper
 
 	discoveryClient discovery.DiscoveryInterface
+	cachedShortcuts []kubectl.ResourceShortcuts
 }
 
 var _ meta.RESTMapper = &shortcutExpander{}
 
-func NewShortcutExpander(delegate meta.RESTMapper, client discovery.DiscoveryInterface) shortcutExpander {
-	return shortcutExpander{RESTMapper: delegate, discoveryClient: client}
+func NewShortcutExpander(delegate meta.RESTMapper, client discovery.DiscoveryInterface) *shortcutExpander {
```

Comment:
I wouldn't expect changes to shortcut expander... suggestions are useful even if you're not using expansion... separate out the suggestion function

Reply:
@anfernee Did you address that comment?

Reply:
yes, i did.